src/main/java/edu/cornell/kfs/coa/document/SubAccountGlobalMaintainableImpl.java

```diff
@@ -101,6 +131,49 @@ private boolean isSubAccountTypeCodeCostShare() {
         return SubAccount.class;
     }
     
+    @Override
+    public void addNewLineToCollection(String collectionName) {
+        super.addNewLineToCollection(collectionName);
+        if (isNewAccountDetailsCollection(collectionName)) {
+            SubAccountGlobal subAccountGlobal = (SubAccountGlobal) getBusinessObject();
+            int newItemIndex = subAccountGlobal.getSubAccountGlobalNewAccountDetails().size() - 1;
+            setSequenceNumbersOnNewAccountDetails(subAccountGlobal, newItemIndex);
+        }
+    }
+    
+    @Override
+    public void addMultipleValueLookupResults(MaintenanceDocument document, String collectionName,
+            Collection<PersistableBusinessObject> rawValues, boolean needsBlank, PersistableBusinessObject bo) {
+        if (isNewAccountDetailsCollection(collectionName)) {
+            SubAccountGlobal subAccountGlobal = (SubAccountGlobal) bo;
+            int oldCollectionSizeAsStartIndex = subAccountGlobal.getSubAccountGlobalNewAccountDetails().size();
+            super.addMultipleValueLookupResults(document, collectionName, rawValues, needsBlank, bo);
+            setSequenceNumbersOnNewAccountDetails(subAccountGlobal, oldCollectionSizeAsStartIndex);
+        } else {
+            super.addMultipleValueLookupResults(document, collectionName, rawValues, needsBlank, bo);
+        }
+    }
+    
+    private void setSequenceNumbersOnNewAccountDetails(SubAccountGlobal subAccountGlobal, int newItemsStartIndex) {
+        List<SubAccountGlobalNewAccountDetail> newAccountDetails = subAccountGlobal.getSubAccountGlobalNewAccountDetails();
+        if (newItemsStartIndex >= newAccountDetails.size()) {
+            return;
+        }
+        
+        long nextSequenceNumber = subAccountGlobal.getNextNewAccountDetailSequenceNumber().longValue();
+        for (int i = newItemsStartIndex; i < newAccountDetails.size(); i++) {
+            SubAccountGlobalNewAccountDetail newAccountDetail = newAccountDetails.get(i);
+            newAccountDetail.setSequenceNumber(Long.valueOf(nextSequenceNumber));
+            nextSequenceNumber++;
+        }
+        
+        subAccountGlobal.setNextNewAccountDetailSequenceNumber(Long.valueOf(nextSequenceNumber));
```

Comment:
I've seen some paradigms in KFS where the get next sequence number function increcements the sequence number in question.  To me that looks cleaner.  I'll leave that up to you if that should be changed

Reply:
In this case the sequence number is coming from a document field instead of a database sequence; however, I'll see if I can rework this to use some kind of next-sequence-number method.